codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieDecoder.java

```diff
@@ -32,27 +32,49 @@
     private final InternalLogger logger = InternalLoggerFactory.getInstance(getClass());
 
     private final boolean strict;
+    private final boolean reportException;
+
+    protected CookieDecoder(boolean strict, boolean reportException) {
+       this.strict = strict;
+       this.reportException = reportException;
+    }
 
     protected CookieDecoder(boolean strict) {
-        this.strict = strict;
+        this(strict, false);
     }
 
     protected DefaultCookie initCookie(String header, int nameBegin, int nameEnd, int valueBegin, int valueEnd) {
         if (nameBegin == -1 || nameBegin == nameEnd) {
-            logger.debug("Skipping cookie with null name");
+            if (logger.isDebugEnabled()) {
```

Comment:
Looks good, but your proposed version introduces computation that is not present in my, or original version as the string will have to be computed, so it will be executed on each request. In my code the JIT will just remove the whole branch and the change will be neutral, do you agree?